solidity/ERC20ForSPL/ERC20ForSPLMintable.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "./openzeppelin-fork/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./openzeppelin-fork/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import './ERC20ForSPLBackbone.sol';
+
+
+/// @title ERC20ForSPLMintable
+/// @author https://twitter.com/mnedelchev_
+/// @notice This contract serves as an interface to the deployed SPLToken on Solana. Through this interface, Ethereum-like address on Neon EVM chain can apply changes on SPLToken account on Solana.
+/// @dev This contract is being used as a BeaconProxy implementation. The Beacon is defined and inherited from ERC20ForSPLBackbone.sol at storage slot 0.
+/// @custom:oz-upgrades-unsafe-allow constructor
+contract ERC20ForSPLMintable is ERC20ForSPLBackbone, Initializable, OwnableUpgradeable {
```

Comment:
Why does `ERC20ForSPLMintable` inherit `OwnableUpgradeable`? The only Factory owner can upgrade the mintable contract as we discussed.

Reply:
We have 2 different owners for this contract. The first owner is DAO (in the future) which can update implementation. And do it through the factory contract. The second owner can mint new tokens but it can't update the code of the contract. Is it really need to use `onlyOwner` and `OwnableUpgradeable` in this case?

Reply:
@s-medvedev I'm just following the logic of the previous ERC20ForSPLMintable standard, please have a look at **[this line](https://github.com/neonlabsorg/neon-evm/blob/develop/solidity/erc20_for_spl.sol#L220)**. Regarding this line in the mintable standard only the contract "admin" _( which is set by the time of the deployment in the constructor )_ has the permissions to mint new tokens. Please tell me should we change it so only the DAO has the possibility to mint new tokens? This is different approach from current mintable erc20forspl.

Reply:
I got it. You are right, only a specific owner can mint new tokens, not the DAO.
I was confused by the name `OwnableUpgradeable` and thought that this base class allows updating the contract. I've looked at the source again and find that all OK.